Marlin/endstops.cpp

```diff
@@ -278,22 +278,22 @@ void Endstops::update() {
     }
   #endif
 
-  #if CORE_IS_XY || CORE_IS_XZ
-    #if ENABLED(COREYX) || ENABLED(COREZX)
-      #define CORE_X_CMP !=
-      #define CORE_X_NOT !
-    #else
-      #define CORE_X_CMP ==
-      #define CORE_X_NOT
-    #endif
+  #if ENABLED(COREXY) || ENABLED(COREXZ)
+    #define CORE_X_CMP ==
+  #elif ENABLED(COREYX) | ENABLED(COREZX)
+    #define CORE_X_CMP !=
+  #endif
     // Head direction in -X axis for CoreXY and CoreXZ bots.
     // If steps differ, both axes are moving.
     // If DeltaA == -DeltaB, the movement is only in the 2nd axis (Y or Z, handled below)
     // If DeltaA ==  DeltaB, the movement is only in the 1st axis (X)
-    if (stepper.current_block->steps[CORE_AXIS_1] != stepper.current_block->steps[CORE_AXIS_2] || stepper.motor_direction(CORE_AXIS_1) CORE_X_CMP stepper.motor_direction(CORE_AXIS_2)) {
-      if (CORE_X_NOT stepper.motor_direction(X_HEAD))
+  #if CORE_IS_XY || CORE_IS_XZ
+    if (stepper.current_block->steps[CORE_AXIS_1] != stepper.current_block->steps[CORE_AXIS_2] || stepper.current_block->steps[CORE_AXIS_1] > 0 && stepper.motor_direction(CORE_AXIS_1) CORE_X_CMP stepper.motor_direction(CORE_AXIS_2)) {
```

Comment:
Added some extra parentheses here to clear up any logical ambiguity, from…
```cpp
if (Q != P || Q > 0 && D1 CORE_X_CMP D2) {
```
…to…
```cpp
if (Q != P || (Q > 0 && D1 CORE_X_CMP D2)) {
```
…even though it just operates like a sieve in forward order.